sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/transformer/ScalaClassSparkDsNTo1Transformer.scala

```diff
@@ -114,11 +122,33 @@ case class ScalaClassSparkDsNTo1Transformer(override val name: String = "ScalaCl
           targetException.setStackTrace(e.getTargetException.getStackTrace ++ e.getStackTrace)
           throw targetException
       }
-    res.asInstanceOf[Dataset[_]].toDF
+
+    val outputClassType = transformMethodInstance.getAnnotatedReturnType.getType.asInstanceOf[ParameterizedTypeImpl].getActualTypeArguments.head.getTypeName
+    val columsFromCaseClass = getCaseClassColumnNames(outputClassType)
+
+    val resAsDF = res.asInstanceOf[Dataset[_]].toDF
+    val resWithSelect = if (outputColumnAutoSelect) resAsDF.select(columsFromCaseClass.map(col): _*) else resAsDF
+
+    if (addPartitionValuesToOutput) {
+      assert(partitionValues.size == 1, s"When using addPartitionValuesToOutput you can only process one partition-key at a time, but ${partitionValues.size} where given: {${partitionValues.mkString(";")}}")
+      partitionValues.head.elements.foldLeft(resWithSelect) {
+        (dataframe, pair) => dataframe.withColumn(pair._1, lit(pair._2))
+      }
+    }
+    else {
+      resWithSelect
+    }
+  }
+
+  private def getCaseClassColumnNames(className: String): Seq[String] = {
+    val outputCaseClass = Environment.classLoader.loadClass(className)
+    outputCaseClass.getDeclaredFields.toSeq.map(_.getName)
```

Comment:
Clean reflection implementation would be
```
import scala.reflect.runtime.universe._
def classAccessors[T: TypeTag]: List[MethodSymbol] = typeOf[T].members.collect {
  case m: MethodSymbol if m.isCaseAccessor => m
}.toList
```
We should add this to ProductUtils, as we need this in several places.

Reply:
Done, but in a diffent pr: https://github.com/smart-data-lake/smart-data-lake/pull/552